insights/tests/client/test_client.py

```diff
@@ -329,10 +330,50 @@ def test_upload_500_retry(_, upload_archive):
 
         upload_archive.assert_called()
         assert upload_archive.call_count == retries
+        logger.error.assert_any_call("Upload attempt %d of %d failed! Reason: %s", 1, config.retries, ANY)
+        logger.error.assert_called_with("All attempts to upload have failed!")
     finally:
         sys.argv = tmp
 
 
+@patch('insights.client.client.constants.sleep_time', 0)
+@patch('insights.client.client.InsightsConnection.upload_archive')
+@patch('insights.client.os.path.exists', return_value=True)
+@patch("insights.client.client.logger")
+def test_upload_exception_retry(logger, path_exists, upload_archive):
```

Comment:
Sometimes a preceding underscore is used to mark an argument with no usage in the function.
```def test_upload_exception_retry(logger, _path_exists, upload_archive):```

Reply:
I removed the mock for os.path.exists, so the path_exists argument wasn't needed anymore.